### PR TITLE
 create person , student and teacher classes

### PR DESCRIPTION
**Pull request by jaamanu**
In this PR i 
Create class Person with the following:
Instance vars: @id, @name, and @age.
Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
Getters for @id, @name, and @age.
Setters for @name and @age.

Create class Student with the following:
Inherits from Person.
Constructor extends parent's constructor by adding @classroom and a parameter for it.
Method play_hooky that returns "¯\(ツ)/¯".
Create class Teacher with the following:
Inherits from Person.
Constructor extends parent's constructor by adding @specialization and a parameter for it.

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @jaamanu 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
invalide

## Invalid Code Review Request

You have submitted a project, that has failing linter checks, Kindly fix the failing linter checks 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  atrr_reader :id, :name, :age
+  atrr_writer :name, :age
```
Comment (julie-ify) [inline person.rb]: - [ ] I think you meant to use `attr_reader`, not `atrr_reader` (notice double 'r').

- [ ] The `:name` and `:age` variables should have getter and setter (attr_accessor), not just setter.

  In conclusion, we should have these:

  ```
  attr_reader :id
  attr_accessor :name, :age
  ```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(age, classroom, name = 'unknown', parent_permission: true)
+    super(name, age, parent_permission)
```
Comment (julie-ify) [inline student.rb]: - [ ] The `parent_permission` parameter uses the colon notation so we should update the assignment in the super method correctly.

  ```
  super(age, name, parent_permission: parent_permission)
  ```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
+  def initialize(age, specialization, name = 'unknown', parent_permission: true)
+    super(name, age, parent_permission)
```
Comment (julie-ify) [inline teacher.rb]: - [ ] The `parent_permission` parameter uses the colon notation so we should update the assignment in the super method correctly.

  ```
  super(age, name, parent_permission: parent_permission)
  ```

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
```
Comment (julie-ify) [inline student.rb]: - [ ] Kindly import the Person class in this file to allow inheritance.

  ```
  require_relative './person.rb'
  ```

File: teacher.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
```
Comment (julie-ify) [inline teacher.rb]: - [ ] Kindly import the Person class in this file to allow inheritance.

  ```
  require_relative './person.rb'
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @jaamanu,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb (hunk 5 above)
Comment (mahbubul14) [reply teacher.rb]: > * [ ]  Kindly import the Person class in this file to allow inheritance.
>   ```
>   require_relative './person.rb'
>   ```

- [ ] As the previous reviewer suggested, kindly add the mentioned line on the `teacher.rb` file. :+1:

File: student.rb (hunk 4 above)
Comment (mahbubul14) [reply student.rb]: > * [ ]  Kindly import the Person class in this file to allow inheritance.
>   ```
>   require_relative './person.rb'
>   ```

- [ ] As the previous reviewer suggested, kindly add the mentioned line on the `student.rb` file. :+1:

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @jaamanu,

Great work on making the changes requested by the previous reviewer 👏

You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

# Required Changes ♻️

- [ ] Kindly check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Proper Github flow used.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "fix bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @jaamanu ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**